envoy/http/async_client.h

```diff
@@ -163,13 +170,52 @@ class AsyncClient {
      */
     virtual void reset() PURE;
 
+    /***
+     * Add destructor callback.
+     */
+    virtual void setDestructorCallback(StreamDestructorCallbacks callback) PURE;
```

Comment:
I think we can be a bit more clear in comments
"Register a destructor to be called on [stream deletion? client deletion? end of stream?]. The callback must persist beyond the lifetime of the stream or be unregistered via removeDestructorCallback.
Is it illegal to call set if something is already set?

Reply:
Clarified semantics.